src/ERC1271.sol

```diff
@@ -0,0 +1,35 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.23;
+
+import {IERC1271} from "./interfaces/IERC1271.sol";
+import {EIP712} from "./EIP712.sol";
+
+/// @title ERC-1271
+///
+/// @notice Abstract ERC1271 implementation protecting against cross account replay attacks.
+///
+/// @author Uniswap
+/// @author Modified from Coinbase (https://github.com/coinbase/smart-wallet)
+/// @author Modified from Solady (https://github.com/vectorized/solady/blob/main/src/accounts/ERC1271.sol)
+abstract contract ERC1271 is IERC1271, EIP712 {
```

Comment:
interesting.. sort of an odd inheritance but I see that it's because you need _hashTypedData?

Can both isValidSignature and _isValidSignature be implemented at the base contract, which then inherits ERC1271 and EIP712?

Reply:
Yea I agree its a bit clunky I like your suggestion!